### PR TITLE
docs: add note about multiple sync options on annotation

### PR DESCRIPTION
**Pull request by QuingKhaos**
I had do dig through the code to see if multiple sync options on the `    argocd.argoproj.io/sync-options` were supported or not and how.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: docs/user-guide/sync-options.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Sync Options
 
-ArgoCD allows users to customize some aspects of how it syncs the desired state in the target cluster. Some Sync Options can defined as annotations in a specific resource. Most of the Sync Options are configured in the Application resource `spec.syncPolicy.syncOptions` attribute.
+ArgoCD allows users to customize some aspects of how it syncs the desired state in the target cluster. Some Sync Options can defined as annotations in a specific resource. Most of the Sync Options are configured in the Application resource `spec.syncPolicy.syncOptions` attribute. Multiple Sync Options which are configured with the `argocd.argoproj.io/sync-options` annotation can be concatenated with a `,` in the annotion value, white spaces will be trimmed.
```
Comment (crenshaw-dev) [inline docs/user-guide/sync-options.md:3]:
```suggestion
Argo CD allows users to customize some aspects of how it syncs the desired state in the target cluster. Some Sync Options can defined as annotations in a specific resource. Most of the Sync Options are configured in the Application resource `spec.syncPolicy.syncOptions` attribute. Multiple Sync Options which are configured with the `argocd.argoproj.io/sync-options` annotation can be concatenated with a `,` in the annotation value; white spaces will be trimmed.
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Thanks @QuingKhaos! Just looking for a couple tweaks.

**Comment by QuingKhaos** [comment]
Thanks @crenshaw-dev, changes are incorporated.

**Review by crenshaw-dev — APPROVED** [review]
Thanks, @QuingKhaos!

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.4 for 2.4.13.

**Pull request merged**